### DISCUSSION TITLE
Change Default Browser

### DISCUSSION THREAD

**Discussion by hansteg** [ttscoff/bunch-docs#166] [Q&A] [answered]
Hi,
is it possible to change the default browser with Bunch? I use different ones for coding and for other tasks. Thanks!
Hans

**Comment by ttscoff** [comment]
Might be possible to add, but I'd have to do some research on it. 
Currently not an option, though Bunch itself can target different 
browsers when opening urls with prefixes like `Firefox:http://...`.

-Brett

**Comment by hansteg** [comment]
Great, thanks!

On 12 Nov 2021, at 7:24, Brett Terpstra wrote:

> Might be possible to add, but I'd have to do some research on it.
> Currently not an option, though Bunch itself can target different
> browsers when opening urls with prefixes like `Firefox:http://...`.
>
> -Brett
>
>
>
> On 11 Nov 2021, at 21:47, hansteg wrote:
>
>> Hi,
>> is it possible to change the default browser with Bunch? I use
>> different ones for coding and for other tasks. Thanks!
>> Hans
>>
>> -- 
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly or view it on GitHub:
>> https://github.com/ttscoff/bunch/discussions/166
>
>
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/ttscoff/bunch/discussions/166#discussioncomment-1631483


---------------------
All the best,

Hans

http://tammen.org

...just another worker in rhythms, frequencies and intensities.

**Comment by ttscoff** [comment, marked as answer]
I did a little looking into this. The way to do it nicely is to use LSSetDefaultHandlerForURLScheme, but since Yosemite, doing that for the `http` protocol triggers a confirmation dialog, which defeats the purpose of automating it. It's a security feature, and bypassing it would get a little dirtier than I like to with Bunch. I'm going to say that the url prefixes will cover most needs in this area, and would recommend an app like Choosy or Browser Fairy for opening different websites with different browsers based on predefined rules.

**Comment by hansteg** [comment]
On 19 Nov 2021, at 14:18, Brett Terpstra wrote:

> I did a little looking into this. The way to do it nicely is to use 
> LSSetDefaultHandlerForURLScheme, but since Yosemite, doing that for 
> the `http` protocol triggers a confirmation dialog, which defeats the 
> purpose of automating it.

I see. Thanks for looking into this!

> It's a security feature, and bypassing it would get a little dirtier 
> than I like to with Bunch.

I can'r really complain about security measures. ;-)

> I'm going to say that the url prefixes will cover most needs in this 
> area, and would recommend an app like Choosy or Browser Fairy for 
> opening different websites with different browsers based on predefined 
> rules.

I didn't know about those and will check them out. They may bring some 
relief to the fact that some websites only work with specific browsers. 
Which seems to have been getting worse over the years.

Thanks!

Hans

**Reply by karkraeg** [reply]
Did you find a way to change the default browser while a bunch is active with any of the mentioned apps?

**Reply by karkraeg** [reply]
I found https://www.felixparadis.com/posts/how-to-set-the-default-browser-from-the-command-line-on-a-mac/ which works nicely. Looking to change the default browser when working 😄
